saas/controllers/main.py

```diff
@@ -21,16 +22,26 @@ def auth_to_build(self, build_id=None):
 
         return env.get_template('saas_auth.html').render({'build_url': build_url})
 
-    @route('/saas/create-fast-build/<int:template_operator_id>', type='http', auth='public')
-    def create_fast_build(self, template_operator_id, **kwargs):
+    @route(['/saas/<model("saas.template"):template_id>/create-fast-build',
+            '/saas/<model("saas.template"):template_id>/<model("saas.template.operator"):template_operator_id>'
+            '/create-fast-build'], type='http', auth='public')
+    def create_fast_build(self, template_id, template_operator_id=None, **kwargs):
         if not kwargs:
             kwargs = {}
-        t_op = request.env['saas.template.operator'].sudo().browse(template_operator_id)
-        if not t_op or t_op.state != 'done':
+        if not template_operator_id:
+            ready_t_ops = request.env['saas.template.operator'].search([
+                ('template_id', '=', template_id.id),
+                ('state', '=', 'done'),
+            ])
```

Comment:
Does that meant, that another method is incorrect?


```
    @api.onchange('random')
    def change_operator(self):
        if self.random:
            random_operator = choice(self.template_id.operator_ids)
            self.template_operator_id = random_operator.id

```

Anyway, such code must be in a single place. You shall not copy-paste it

Reply:
I decided that it would be incorrect to use the method from wizard. In the context of use in wizard there is a domain which filters not ready operators. I think that I can write a more universal method for saas.template.operator model  that will return a ready random operator

Reply:
ok